docs/adrs/0010-jetstream-shared-schema.md

```diff
@@ -0,0 +1,155 @@
+* [**Option B**] `experimenter` (Nimbus) repository
+* [**Option C**] `nimbus-shared` repository
+
+
+## Decision Outcome
+
+Chosen option: **Option 1B**: define the schema using Pydantic, and put this definition in the `experimenter` repository. **Option 1** Pydantic was chosen because it meets the needs of the solution, and fits best with the language and approaches in use by the relevant systems. **Option B** `experimenter` repo was chosen as the location for the schema because it represents the closest thing to a monorepo (or at least somewhat consolidated location for shared content), and any schemas shared among Nimbus systems will likely be used in the `experimenter` repo.
+
+### Overview of Solution Integration
+![Architecture of chosen solution, showing location of schema and how it works with existing systems](./images/jetstream-schema-arch.png)
+
+See below for more info on the options and their pros/cons.
+
+### Positive Consequences
+
+* Ability to generate and ingest analysis results in a structured way
+* Nimbus repository as central source of truth
+  * Nimbus repository may become home to Jetstream in the future
+  * Other Nimbus schemas can live in the same place
+* Aligns with FastAPI approach used for ongoing Cirrus work
+* Easily generate schema definitions for different scenarios
+  * Pydantic allows us to automatically generate bindings and publish new versions to external repositories (e.g., PyPi for use by Jetstream)
+  * Django, Typescript, JSON
+* Publishing the schema as a library handles concerns about schema changes and synchronization
+  * Schema versioning is basically free via the published library
+  * Ability to reference older versions of the schema and tag results with the relevant version
+
+### Negative Consequences
+
+* Could be confusing to have the schema definition not live where the data it defines is generated
+  * And the definition does not live with the data
+* Need to consume a new dependency in order to use the schema
+
+### Prior Art
+
+This solution is similar in practice to how [Protocol Buffers](https://protobuf.dev/overview/) (also known as Protobuf) are used to document schemas. For example, Google uses them extensively as a data format that provides typed schemas and serialization for structured data. Our solution could be used in a similarly broad fashion, and Protobufs would be applicable for our use-case, but Pydantic provides more familiarity to a team and organization with much greater usage of Python (in particular, both Experimenter and Jetstream).
+
+## Pros and Cons of the Schema Definition Options
+Chosen option is marked in **bold**.
+
+### **[Option 1] Pydantic**
+
+* Create schema definitions using Pydantic
+* Autogenerate bindings and publish to, e.g., pypi/npm/cargo for various use-cases
+
+#### Pros
+
+* Aligns with FastAPI approach used for ongoing Cirrus work
+* Easily generate schema definitions for different scenarios
+  * Django, Typescript, JSON
+
+#### Cons
+
```

Comment:
Another con I can think is if the consuming applications are not using the same version